Restructure

razage opened the pull request:
Nobody asked for this, but I think it's a good idea.

razage committed:
huge sweeping restructure of the project.

even though we're ignoring standard python conventions by using camel case, I feel like standard python organization shouldn't be much of an issue. I've moved everything into packages and made a base level package. I've moved several loose functions to a helpers file and the custom classes to the Core package. I made an override to the load_extension method to make this move not be a hindrance. this does present some issues if we plan on there being external cogs, but that is addressable as well.

razage committed:
moved some functions to helper files and fixed a ton of PEP-8 violations.

I had to add bot as a parameter to is_owner since I moved it. It might be better to just make this a class method. I also added ffmpeg to the gitignore so I can click stage all at long last.

crazybmanp commented:
```
> 1) i feel like we should figure out if we can move extension loading out of the run.py
> 2. i don't like using helpers for functions that are tied so close to specific functionality, (specifically the memes helper, it can't be reused in any context) i also feel like Util would be a more standard name, isn't it?
> 3: as said, we should look into moving extension cogs back out of RyuZU since they should work the same as any jackass that wants to add whatever to RyuZU. especially since the OWext cog is not up to the standards of the system and is not representitive of the quality of the rest of the bot.
> 4] i feel like the core.init.py file is way too important with both the bot and cog defenitions, but i don't know what we would do without making it more difficult to use those classes (maybe we just do it any way)
> *extra note on 2: the core helper makes sense, that is a completely reusable method
> 5> the TEMPLATE.py file is old, outdated and crap, we should just delete it
```

razage committed:
moved the cog loading code out of the run file and moved the cogs out of the base package.

cogs can now be considered external. if there is an internal cog, just pass core_cog=True to the load_extensions method.

razage committed:
moved helpers to utils and removed memes helper.

i don't love this, but it was a pretty useless helper file anyway. utils and helpers are both commonly used in python, so either works. but i figure we'll go with utils instead.

razage committed:
removed template file.

this had redundant parenthesis to begin with, but it's also invalid now. i don't know that it ever served much purpose anyway, so rather than update it, i'm just going to remove it.

razage commented:
I've made most of the adjustments you mentioned, but I feel like your concern for the Core package is irrelevant. It is named Core precisely because it is important. I feel that getting both of them in one place makes sense with our current needs, and that it's a much better alternative than having Core.Bot.Bot or from Core.Bot import Bot.

crazybmanp reviewed (approved):
I feel like in the future we will have to see if there is a way to define the bot and cog classes in the init of core but have thier functions defined elsewhere. They will likely get to big for that file.
